shelley/chain-and-ledger/executable-spec/src/TxData.hs

```diff
@@ -247,16 +275,25 @@ instance forall crypto
   => Ord (WitVKey crypto) where
     compare = comparing witKeyHash
 
+newtype Vlds crypto = Vlds (Map (ScriptHash crypto) (Script crypto))
+  deriving (Show, Eq, Generic, NoUnexpectedThunks, ToCBOR, FromCBOR)
+
 -- |A fully formed transaction.
 data Tx crypto
   = Tx
       { _body           :: !(TxBody crypto)
-      , _witnessVKeySet :: !(Set (WitVKey crypto))
-      , _witnessMSigMap ::
-          Map (ScriptHash crypto) (MultiSig crypto)
+      , _unsignedData   :: !(UnsignedData crypto)
+      } deriving (Show, Eq, Generic)
+
+data UnsignedData crypto
+  = UnsignedData
+      { _witnessVKeySet :: !(Set (WitVKey crypto))
+      , _txvlds         :: Vlds crypto
+      , _txdats         :: Map (DataHash crypto) (Data crypto)
```

Comment:
is this meant to be the mapping from script hashes to scripts ?
I think the idea of integration of new witness types (which plutus would be I think) was to add another witness type, probably in the form of an explicit map.

Reply:
This is the witnesses - VKey and Validators (`Vlds`, which is indeed a map of hashes to scripts). `txdats` are the inputs to the plutus validators (i.e. the map from their hashes to the data)